src/components/MovieGallery/MovieGallery.tsx

```diff
@@ -1,19 +1,54 @@
+import { useEffect } from 'react'
 import style from './Moviegallery.module.scss'
 import MovieCard from '../MovieCard/MovieCard'
-import { information } from '../../data/data.js'
-
+import { useAppSelector } from '../../hooks/useTypeRedux'
+import { useDispatch } from 'react-redux'
+import { fetchData } from '../../store/actions/data'
 
 function MovieGallery() {
-  const movieItems = information.map((item) => (
+
+  const { data, loading, error } = useAppSelector(state => state.data)
+
+  const sort = useAppSelector(state => state.sort)
+
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    dispatch(fetchData(sort))
+  }, [sort])
+
```

Comment:
you need to handle this better. You can dispatch a sort action with the sort value where the sort happens